assignments/02-cli-constructive.md

```diff
@@ -55,63 +56,101 @@ $ touch rng.c
 $ ./build.sh 
 cc -c rng.c
 cc -o foo foo.o rng.o
+$ ./foo 
+hello, world! your randomly chosen number is 4
 $
 ```
 
-Notice two things:
+Notice three things:
 
 * We can run `build.sh` and if nothing needs to get rebuilt, nothing will be
   rebuilt
-* We can update individual C files without recompiling other C files
-
-...
+* We can update individual C files without recompiling other C files (we do
+  need to link them at the end, though)
+* By "changed", we mean that the file's modification time is newer than its
+  target; the file contents may not have changed
+
+This is a pretty standard set of features for build systems. Many people have
+put many years into making build systems great. Because we are dealing with
+small code and don't have a lot of strange requirements, you are only going to
+peek a little bit into that yawning abyss. We hope that after writing your
+build script, you look at Make with a little less apprehension. You too could
+write such a tool given enough time.
+
+Unlike other build systems, you are going to put the "should we recompile this"
+logic in the same file as the project-specific components. Together, they will
+form the build script for `foo`.
 
 ### Requirements
 
+This all might sound a little bit scary but we promise that explicitly laying
+out the behavior like this makes for easier implementation. You can think of it
+like a checklist.
+
 > Note: The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
 > "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this
 > document are to be interpreted as described in [RFC
 > 2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
 You MUST:
 
+* Be able to build `foo` by running `./build.sh`
+* Only recompile a target if any of its dependencies' modified-time (mtime) is
```

Comment:
I think this requirement (rebuilding based on modification time) warrants a paragraph in the text above. We are not yet at BLD and so the fact that build systems work like this won't be evident to many students.

Reply:
Also maybe rephrase as "if at least one of its dependencies has been modified more recently than it"

Reply:
Added note about mtime